lib/explorer/shared.ex

```diff
@@ -366,6 +387,14 @@ defmodule Explorer.Shared do
   def merge_numeric_dtype({:f, left}, {:f, right}), do: {:f, max(left, right)}
   def merge_numeric_dtype({:f, _} = float, :null), do: float
   def merge_numeric_dtype(:null, {:f, _} = float), do: float
+
+  def merge_numeric_dtype({:decimal, _, _} = decimal, :null), do: decimal
+  def merge_numeric_dtype(:null, {:decimal, _, _} = decimal), do: decimal
+
+  # For now, float has priority over decimals due to Polars.
```

Comment:
I think this makes sense in general. Float is imprecise, so you by definition go imprecise once you mix precise (decimal) and imprecise (float).